src/DropdownMenu/DropdownMenu.tsx

```diff
@@ -0,0 +1,158 @@
+  selectedItem,
+  setSelectedItem,
+  ...listProps
+}: DropdownMenuProps): JSX.Element {
+  const anchorRef = useRef<HTMLElement>(null)
+  const [overlayRef, updateOverlayRef] = useRenderForcingRef<HTMLDivElement>()
+
+  const anchorId = `dropdownMenuAnchor-${randomId()}`
+
+  const [open, setOpen] = useState<boolean>(false)
+  const [focusType, setFocusType] = useState<null | 'anchor' | 'list'>(null)
+  const onDismiss = useCallback(() => {
+    setOpen(false)
+    setFocusType(null)
+  }, [])
+
+  const onAnchorKeyDown = useCallback(
+    (event: React.KeyboardEvent<HTMLElement>) => {
+      if (!event.defaultPrevented) {
+        if (!open) {
+          if (event.key === 'ArrowDown' || event.key === 'ArrowUp') {
+            setFocusType('list')
+            setOpen(true)
+            event.preventDefault()
+          } else if (event.key === ' ' || event.key === 'Enter') {
+            setFocusType('anchor')
+            setOpen(true)
+            event.preventDefault()
+          }
+        } else if (focusType === 'anchor') {
+          if (['ArrowDown', 'ArrowUp', 'Tab', 'Enter'].indexOf(event.key) !== -1) {
+            setFocusType('list')
+            event.preventDefault()
+          } else if (event.key === 'Escape') {
+            onDismiss()
+            event.preventDefault()
+          }
+        }
+      }
+    },
+    [open, focusType, onDismiss]
+  )
+  const onAnchorClick = useCallback(
+    (event: React.MouseEvent<HTMLElement>) => {
+      if (!event.defaultPrevented && event.button === 0 && !open) {
+        setOpen(true)
+        setFocusType('anchor')
+      }
+    },
+    [open]
+  )
+
+  const {position} = useAnchoredPosition({anchorElementRef: anchorRef, floatingElementRef: overlayRef})
+
+  useFocusZone({containerRef: overlayRef, disabled: !open || focusType !== 'list' || !position})
+  useFocusTrap({containerRef: overlayRef, disabled: !open || focusType !== 'list' || !position})
+
+  return (
+    <>
+      {renderAnchor({
+        ref: anchorRef,
+        id: anchorId,
+        'aria-labelledby': anchorId,
+        'aria-haspopup': 'listbox',
+        children: selectedItem?.text ?? placeholder,
+        onClick: onAnchorClick,
+        onKeyDown: onAnchorKeyDown
+      })}
+      {open && (
+        <Overlay
+          initialFocusRef={anchorRef}
+          returnFocusRef={anchorRef}
+          onClickOutside={onDismiss}
+          onEscape={onDismiss}
+          ref={updateOverlayRef}
+          {...position}
+        >
+          <List
+            {...listProps}
+            role="listbox"
+            renderItem={({onClick, onKeyDown, item, ...itemProps}) => {
+              const itemActivated = () => {
+                setSelectedItem?.(item === selectedItem ? undefined : item)
+                onDismiss()
+              }
+
+              return renderItem({
+                ...itemProps,
+                item,
+                role: 'option',
+                selected: item === selectedItem,
```

Comment:
Do you think we might need a more robust equality check of objects? (for example, lodash's [`isEqual`](https://lodash.com/docs/4.17.15#isEqual) function)

Reply:
This was an intentional design decision to avoid any complications with duplicate or similar items.  We talked about comparing `text`, but that would be very easy to have duplicates.  We could also require an `id` prop on each item, but that's very restrictive.  In the end, the only way to ensure we have the _exact_ item selected is to do an object equality check.  Happy to hear other ideas if you have them.

Reply:
That's fine with me. Just wanted to call attention to it in case it was unintentional :)